Source/JavaScriptCore/wasm/WasmFormat.h

```diff
@@ -122,6 +122,12 @@ inline bool isI31ref(Type type)
     return isRefType(type) && type.index == static_cast<TypeIndex>(TypeKind::I31ref);
 }
 
+inline bool isArrayref(Type type)
+{
+    ASSERT(Options::useWebAssemblyGC());
```

Comment:
Yes, I think you're right that this is incorrect. I think the assertion in the I31ref function above that is incorrect too, and causes some issues in a debug test run for the `call_ref` tests (because typed function refs are turned on, but GC is turned off).

Instead, it should probably always return `false` when GC is off.